### PR TITLE
Feature 3.41 grades

### PR DESCRIPTION
**Pull request by denisfruza**
https://trello.com/c/CACNkHrc

### CODE REVIEW

File: backend/src/main/java/com/pharmacySystem/service/implementations/DermatologistService.java [hunk 1]
```diff
@@ -13,9 +14,15 @@
...
+	@Autowired
+	private GradeRepository gradeRepository;
+	
 	@Override
 	public Dermatologist findById(Long id) {
-		return dermatologistRepository.findById(id).orElse(null);
+		Dermatologist dermatologist = dermatologistRepository.findById(id).orElse(null);
+		dermatologist.setAverageGrade(gradeRepository.findAverageGradeByGradedIdAndGradeType(dermatologist.getId(), GradeType.DERMATOLOGIST_GRADE));
+		
+		return dermatologist;
```
Comment (dbulaja98) [inline backend/src/main/java/com/pharmacySystem/service/implementations/DermatologistService.java:25]: Method name is findById, but you also set average grade, so maybe it is smart to change method name.

File: backend/src/main/java/com/pharmacySystem/service/implementations/GradeService.java [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+	@Override
+	public Set<EmployeeGradeDTO> findDermatologistsForGradingByPatientId(Long id) {
+		Set<Appointment> pastAppointments = appointmentRepository.findPastAppointmentsByPatientId(id, AppointmentType.DERMATOLOGIST_APPOINTMENT);
+		Set<EmployeeGradeDTO> employeeGrades = new HashSet<EmployeeGradeDTO>();
+		for(Appointment a : pastAppointments) {
+			Dermatologist dermatologist = dermatologistService.findById(a.getExaminerId());
+			EmployeeGradeDTO employeeGrade = GradeMapper.createEmployeeGradeDTOfromDermatologist(dermatologist);
+			getPreviousEmployeeGrade(employeeGrade, id, dermatologist.getId(), GradeType.DERMATOLOGIST_GRADE);
+			if(!employeeAlreadyAdded(employeeGrades, employeeGrade.getEmployeeId()))
```
Comment (dbulaja98) [inline backend/src/main/java/com/pharmacySystem/service/implementations/GradeService.java:92]: From*

File: backend/src/main/java/com/pharmacySystem/service/implementations/GradeService.java [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+	
+
+	@Override
+	public Set<MedicineGradeDTO> findMedicinesForGradingByPatientId(Long id) {
+		Set<Medicine> medicines = new HashSet<Medicine>();
+		Set<MedicineGradeDTO> medicineGradeDTOs = new HashSet<MedicineGradeDTO>();
+		getMedicinesFromReservations(id, medicines);
+		getMedicinesFromPrescriptions(id, medicines);
+		for(Medicine medicine : medicines) {
+			MedicineGradeDTO medicineGradeDTO= GradeMapper.createMedicineGradeDTOfromMedicine(medicine);
```
Comment (dbulaja98) [inline backend/src/main/java/com/pharmacySystem/service/implementations/GradeService.java:107]: Same

File: backend/src/main/java/com/pharmacySystem/service/implementations/MedicineService.java [hunk 4]
```diff
@@ -13,9 +15,24 @@
...
+			medicine.setAverageGrade(gradeRepository.findAverageGradeByGradedIdAndGradeType(medicine.getId(), GradeType.MEDICINE_GRADE));
+		}
+		return medicines;
+	}
+
+	@Override
+	public Medicine findById(Long id) {
+		Medicine medicine = medicineRepository.findById(id).orElse(null);
+		medicine.setAverageGrade(gradeRepository.findAverageGradeByGradedIdAndGradeType(medicine.getId(), GradeType.MEDICINE_GRADE));
+
```
Comment (dbulaja98) [inline backend/src/main/java/com/pharmacySystem/service/implementations/MedicineService.java:34]: Same as first

File: backend/src/main/java/com/pharmacySystem/service/implementations/PharmacistService.java [hunk 5]
```diff
@@ -12,8 +15,14 @@
...
+	@Autowired
+	private GradeRepository gradeRepository;
+	
 	@Override
 	public Pharmacist findById(Long id) {
-		return pharmacistRepository.findById(id).orElse(null);
+		Pharmacist pharmacist = pharmacistRepository.findById(id).orElse(null);
+		pharmacist.setAverageGrade(gradeRepository.findAverageGradeByGradedIdAndGradeType(pharmacist.getId(), GradeType.PHARMACIST_GRADE));
+		
+		return pharmacist;
```
Comment (dbulaja98) [inline backend/src/main/java/com/pharmacySystem/service/implementations/PharmacistService.java:26]: :D

File: backend/src/main/java/com/pharmacySystem/service/implementations/GradeService.java (hunk 2 above)
Comment (denisfruza) [reply backend/src/main/java/com/pharmacySystem/service/implementations/GradeService.java:92]: fixed

File: backend/src/main/java/com/pharmacySystem/service/implementations/MedicineService.java (hunk 4 above)
Comment (denisfruza) [reply backend/src/main/java/com/pharmacySystem/service/implementations/MedicineService.java:34]: fixed

File: backend/src/main/java/com/pharmacySystem/service/implementations/DermatologistService.java (hunk 1 above)
Comment (denisfruza) [reply backend/src/main/java/com/pharmacySystem/service/implementations/DermatologistService.java:25]: Well average grade is always needed when we are getting medicines, pharmacists and pharmacies from database, maybe it's a good idea to change the name

**Review by cakareli — APPROVED** [review]

**Pull request merged**